tests/integration/__tests__/infer-plugins-ext-dir.js

```diff
@@ -0,0 +1,64 @@
+describe("infer file ext that supported by only plugins", () => {
+  describe("basic", () => {
+    runCli("cli/infer-plugins-ext-dir/", ["--write", "src"]).test({
+      status: 0,
+      stdout: "src/file.foo 0ms",
+      write: [
+        {
+          content: "!contents\n",
+          filename: "src/file.foo",
+        },
+      ],
+    });
+  });
+
+  describe("with config option", () => {
+    runCli("cli/infer-plugins-ext-dir-with-config/", [
+      "--config",
+      "foo.mjs",
+      "--write",
+      "src",
+    ]).test({
+      status: 0,
+      stdout: "src/file.foo 0ms",
+      write: [
+        {
+          content: "!contents\n",
+          filename: "src/file.foo",
+        },
+      ],
+    });
+  });
+
+  describe("with overrides options", () => {
+    runCli("cli/infer-plugins-ext-dir-with-overrides/", [
+      "--write",
+      "src",
+    ]).test({
+      status: 0,
+      stdout: "src/file.foo 0ms",
+      write: [
+        {
+          content: "!contents\n",
+          filename: "src/file.foo",
+        },
+      ],
+    });
+  });
+
+  describe("with defaultOptions", () => {
+    runCli("cli/infer-plugins-ext-dir-with-default-options/", [
+      "--write",
+      "src",
+    ]).test({
+      status: 0,
+      stdout: "src/file.foo 0ms",
+      write: [
+        {
+          content: '{"tabWidth":2,"bracketSpacing":false}',
```

Comment:
This should be `"tabWidth": 8`...

Reply:
This is resolved by running with `--no-editorconfig`. [c41424d](https://github.com/prettier/prettier/pull/15155/commits/c41424d1ffadc76b11861c171c060f8388a64c1d)

Reply:
@fisker Does this address your concerns about defaultOptions ?

Reply:
I think if we add `plugins` in config root, it will be 8 too when formatting .js files.

But I'm not sure how will it work when the `plugins` is in `overrides`.( Also concern about other files, not `.foo` files.)